platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileWriter.java

```diff
@@ -154,7 +147,14 @@ public static void writeSignedStateFilesToDirectory(
         Objects.requireNonNull(directory);
         Objects.requireNonNull(signedState);
 
-        writeStateFile(directory, signedState);
+        MerkleRoot state = signedState.getState();
+        if (state.isMutable()) {
```

Comment:
Hmm, are we sure this is the right way to go? Would it make more sense to throw if you try writing a mutable state to disk?

Reply:
No, I'm not so sure if it's the right way to go. The absence of this code causes a bunch of tests to fail. But fixing those tests may be the right thing to do, as we should not allow snapshots creation using mutable state.

Reply:
@lpetrovic05 what do you think? Is this approach justified? That is, we can force the state to become immutable, if it comes to creation of a snapshot

Reply:
I think that Austin is right. We should not silently create a copy of a state, this can cause all sorts of issues like memory leaks and caller expecting his state to remain mutable. I think fixing those tests should be the way to go.
Also, why is it a problem to write a mutable state to disk?

Reply:
>Also, why is it a problem to write a mutable state to disk?

The state is supposed to be hashed first, as far as I understand. And we can hash only an immutable state.